src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -535,16 +535,44 @@ module Make (Inputs : Inputs_intf.S) = struct
 
     let location_of_account_batch t account_ids =
       assert_is_attached t ;
-      let found_locations, leftover_account_ids =
-        List.partition_map account_ids ~f:(fun account_id ->
-            match self_find_location t account_id with
-            | Some location ->
-                Either.first (account_id, Some location)
-            | None ->
-                Either.second account_id )
+      let account_ids_with_locations_rev, leftover_account_ids_rev =
```

Comment:
I suggest using the [version](https://github.com/MinaProtocol/mina/commit/8a5022369d5ae850cd882c3c681bd129eb387822)

Reply:
I took your version, with some minor modifications (your version was reversing the resulting list, and had a some misnomers about the parent always being a database).